### PR TITLE
azuredisk-csi-driver: snapshots feature

### PR DESCRIPTION
**Pull request by dkoshkin**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:

Just like in [AWS EBS CSI driver](https://github.com/mesosphere/kubernetes-base-addons/blob/c979876c422d3a1ec59c599b629c538b636e030b/addons/awsebscsiprovisioner/0.5.x/awsebscsiprovisioner-4.yaml#L32) and [GCP CSI driver](https://github.com/mesosphere/charts/blob/8d770e4bcecea8c8bda77df46dc2cad6d8311fea/stable/gcpdisk-csi-driver/templates/csi-gcpdisk-snapshot-controller.yaml#L1) enable the Snapshot feature by default.

The chart already has [an option for this](https://github.com/mesosphere/charts/blob/8d770e4bcecea8c8bda77df46dc2cad6d8311fea/stable/azuredisk-csi-driver/templates/csi-snapshot-controller.yaml#L1) , here just changing the default.

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-70322

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
azuredisk-csi-driver: enable the Snapshot controller
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Comment by dkoshkin** [comment]
Tested by deploying a cluster with this branch
```
➜ kubectl api-resources | grep volumesnapshotclasses
volumesnapshotclasses                                                          snapshot.storage.k8s.io                        false        VolumeSnapshotClass
```

File: addons/azuredisk-csi-driver/0.7.x/azuredisk-csi-driver-4.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  name: azuredisk-csi-driver
+  labels:
+    kubeaddons.mesosphere.io/name: azuredisk-csi-driver
+    kubeaddons.mesosphere.io/provides: csi-driver
+  annotations:
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.7.1-2"
```
Comment (hectorj2f) [inline addons/azuredisk-csi-driver/0.7.x/azuredisk-csi-driver-4.yaml:10]: should not it be `0.7.2-4` ?

File: addons/azuredisk-csi-driver/0.7.x/azuredisk-csi-driver-4.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    appversion.kubeaddons.mesosphere.io/azuredisk-csi-driver: "0.7.1"
```
Comment (hectorj2f) [inline addons/azuredisk-csi-driver/0.7.x/azuredisk-csi-driver-4.yaml:11]: also `0.7.2` here!

**Comment by dkoshkin** [comment]
> Keep in mind that this driver is one of the few addons left that are NOT tested automatically. This LGTM but let's make sure we put it through some manual testing before we merge.

👍 yep tested locally with the latest changes

```
➜  dkkonvoyazure cat cluster.yaml | grep -B 1 configVersion
    - configRepository: https://github.com/mesosphere/kubernetes-base-addons
      configVersion: dkoshkin-azure-csi-snapshots
```

```
Kubernetes cluster and addons deployed successfully!

Run `konvoy apply kubeconfig` to update kubectl credentials.

Run `konvoy check` to verify that the cluster has reached a steady state and all deployments have finished.

Navigate to the URL below to access various services running in the cluster.
  https://52.247.16.196/ops/landing
And login using the credentials below.
  Username: elegant_feynman
  Password: gyyVa6B3rh5QrYpguuDorLby9PX9d24lPROfohMzYGFkLupUHRyKJ1E88YnyqLIV

If the cluster was recently created, the dashboard and services may take a few minutes to be accessible.
```

```
➜  dkkonvoyazure k api-resources | grep volumesnapshotclasses
volumesnapshotclasses
```

**Pull request merged**